src/foam/nanos/notification/NotificationSetting.js

```diff
@@ -128,27 +129,31 @@ foam.CLASS({
       name: 'authorizeOnCreate',
       javaCode: `
       AuthService auth = (AuthService) x.get("auth");
+      if ( ! checkSpid(x) ) throw new AuthorizationException(LACKS_READ_PERMISSION);
```

Comment:
Perhaps the message can be 'read or modify', then it will work for all scenarios.